src/lib/structures/Josh.ts

```diff
@@ -96,6 +96,8 @@ export class Josh<StoredValue = unknown> {
    * ```
    */
   public async init(): Promise<this> {
+    for (const middleware of this.middlewares.values()) middleware.init(this.middlewares);
```

Comment:
Shouldn't this init be awaited in case it requires async?

Reply:
This is possible, I guess I ought to add that in case.

Reply:
Actually to follow up on this, having an asynchronous `init()` function would break the `Josh#use()` method making it asynchronous.